src/index.ts

```diff
@@ -28,9 +28,19 @@ async function main() {
 
   if (stampsConfig) {
     logger.debug('stamps config', stampsConfig)
-    const stampManager = new StampsManager()
-    logger.info('starting postage stamp manager')
-    stampManager.start(stampsConfig)
+    let stampManager: HardcodedStampsManager | AutoBuyStampsManager | ExtendsStampManager
```

Comment:
Here also please use the `StampManager` interface. You can then call the `start()` after the `if/else` Stamp Manager initialization block.